modules/lmdbbackend/lmdbbackend.cc

```diff
@@ -1665,7 +1669,27 @@ void LMDBBackend::getAllDomains(vector<DomainInfo>* domains, bool /* doSerial */
     }
 
     di.backend = this;
-    domains->push_back(di);
+    if (zonemap.count(di.zone) == 1) {
+      dups.insert(di.zone);
+    }
+    else {
+      zonemap[di.zone] = di;
+    }
+  }
+
+  for (auto zone : dups) {
+    DomainInfo di;
+
+    if (!(di.id = txn.get<0>(zone, di))) {
```

Comment:
Not sure if if I understand this correctly, but does this not just check if a reverse index entry is present, instead of checking if this is the active one?

Reply:
`.get` calls `.get_multi` passing `true` to the new `onlyOldest` argument

Reply:
this now has comments